tests/testthat/error/test-conditions.txt

```diff
@@ -111,3 +111,19 @@ ordered cast failures mention conversion
 > vec_cast(ordered("x"), ordered("y"))
 Error: Can't convert <ordered<5a425>> to <ordered<df698>>.
 
+
+incompatible size errors
+========================
+
+> stop_incompatible_size(1:2, 3:5, 2L, 3L, x_arg = "", y_arg = "")
+Error: Can't recycle size 2 to match size 3.
```

Comment:
And if there's no name of the LHS, maybe this should be: "Can't recycle vector of size 2"?

Reply:
Did you mean to not mention the target size?

Reply:
That might imply that when you do have a LHS name, you don't need parenthesis

```
Can't recycle vector of size 2 to size 3.

Can't recycle `foo` of size 2 to size 3.

Can't recycle vector of size 2 to match `foo` of size 3.

Can't recycle `foo` of size 2 to match `bar` of size 3.
```

Then you generally just switch "vector" for the name, and maybe add "match"

Reply:
I like it.

Reply:
Oh, maybe a slightly different message depending on if you are using `vec_recycle()` VS `vec_recycle_common()`/`vec_size_common()`. `vec_recycle()` just wouldn't mention the "to match" part since you are just recycling to a specific size.

```r
vec_recycle(x, 3)
#> Can't recycle vector of size 2 to size 3.

vec_recycle(x, 3, x_arg = "foo")
#> Can't recycle `foo` of size 2 to size 3.

# hypothetical, see #1000 
vec_size2(1:2, 1:3)
vec_recycle2(1:2, 1:3)
#> Can't recycle vector of size 2 to match vector of size 3.

vec_size2(1:2, 1:3, x_arg = "foo")
vec_recycle2(1:2, 1:3, x_arg = "foo")
#> Can't recycle `foo` of size 2 to match vector of size 3.

vec_size_common(1:2, 1:3)
vec_recycle_common(1:2, 1:3)
#> Can't recycle `..1` of size 2 to match `..2` of size 3.

vec_size_common(foo = 1:2, 1:3)
vec_recycle_common(foo = 1:2, 1:3)
#> Can't recycle `foo` of size 2 to match `..2` of size 3.

vec_size_common(foo = 1:2, bar = 1:3)
vec_recycle_common(foo = 1:2, bar = 1:3)
#> Can't recycle `foo` of size 2 to match `bar` of size 3.
```

Reply:
I don't love "`foo` of size 2". If you don't want parentheses, I think "with" would be better than "of".